docs/quickstart.rst

```diff
@@ -144,3 +144,140 @@ Otherwise, models are represented as dicts.
+                "name": "date",
+                "type": "string"
+              }
+            ],
+            "operationId": "echo_date",
+            "responses": {
+              "200": {
+                "$ref": "#/responses/200_ok"
+              }
+            }
+          }
+        },
+        "/echo_date_time": {
+          "post": {
+            "description": "Echoes the input date-time into response body",
+            "operationId": "echo_date_time",
+            "parameters": [
+              {
+                "format": "date-time",
+                "in": "formData",
+                "name": "date_time",
+                "type": "string"
+              }
+            ],
+            "responses": {
+              "200": {
+                "$ref": "#/responses/200_ok"
+              }
+            }
+          }
+        }
+      },
+      "responses": {
+          "200_ok": {
+            "description": "HTTP/200 OK",
+            "schema": {
+            "$ref": "#/definitions/json_object"
+          }
+      },
+      "swagger": "2.0"
+    }
+
+and that your views are defined as
+
+.. code-block:: python
+
+    @view_config(route_name='echo_date', renderer='pyramid_swagger')
+    def echo_date(request):
+        input_date = request.swagger_data['date']
+        assert isinstance(input_date, datetime.date)
+        return {'date': input_date}
+
+    @view_config(route_name='echo_date_time', renderer='json')
+    def echo_date(request):
+        input_date_time = request.swagger_data['date_time']
+        assert isinstance(input_date_time, datetime.datetime)
+        return {'date-time': input_date_time}
+
+Calling the ``/echo_date`` endpoint you will receive, as expected, a response that will look like ``{"date": "2017-09-06"}``, while calling ``/echo_date_time`` endpoint you will receive an unexpected HTTP/500 as response.
+
+This happens because ``pyramid_swagger`` renderer performs the proper object unmarshalling (transforming the ``datetime.date`` object to an ISO 8601 string representation) before calling the base JSON renderer; while the default ``json`` renderer, which is not capable to transform to JSON a ``datetime.datetime`` object, will raise an exception.
+
+To workaround the limitation introduced by ``json`` formatter the developer has to *manually*  convert the ``datetime.datetime`` object to the desired string representation or adding `adapters <https://docs.pylonsproject.org/projects/pyramid/en/latest/narr/renderers.html#using-the-add-adapter-method-of-a-custom-json-renderer>`_ for the types not handled by ``json`` renderer.
+Either approaches are introducing an additional complexity on the service developer since he/she has to care about response formatting and could not be compliant with the service specification.
+
+In order to increase the flexibility of the new rendering feature, it is exposed ``PyramidSwaggerRendererFactory`` class which will allow you to define your own custom renderer.
```

Comment:
I think this example isn't super clear still on what my `__call__` method is operating on. Maybe a toy example so readers can see what a "marshalled response value" looks like vs what their `__call__` method might be doing?

Reply:
I'll add an example with a test view that returns data and I'll add comments explaining that the content of the variables on each stage, I'll post the new iterations while the other comments are clear and addressed